bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/ViessmannBridgeHandler.java

```diff
@@ -0,0 +1,337 @@
+/**
+ * Copyright (c) 2021 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.binding.viessmann.internal.handler;
+
+import static org.smarthomej.binding.viessmann.internal.ViessmannBindingConstants.*;
+
+import java.time.Instant;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smarthomej.binding.viessmann.internal.ViessmannDiscoveryService;
+import org.smarthomej.binding.viessmann.internal.api.ViessmannApi;
+import org.smarthomej.binding.viessmann.internal.config.BridgeConfiguration;
+import org.smarthomej.binding.viessmann.internal.dto.device.DeviceDTO;
+import org.smarthomej.binding.viessmann.internal.dto.device.DeviceData;
+import org.smarthomej.binding.viessmann.internal.dto.features.FeatureDataDTO;
+import org.smarthomej.binding.viessmann.internal.dto.features.FeaturesDTO;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link ViessmannBridgeHandler} is responsible for handling the api connection.
+ *
+ * @author Ronny Grun - Initial contribution
+ */
+@NonNullByDefault
+public class ViessmannBridgeHandler extends BaseBridgeHandler {
+    private static final int DEFAULT_API_TIMEOUT_SECONDS = 20;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final HttpClient httpClient;
+
+    private @NonNullByDefault({}) ViessmannApi api;
+    private @NonNullByDefault({}) String apiKey;
+    private @NonNullByDefault({}) String user;
+    private @NonNullByDefault({}) String password;
+    private @NonNullByDefault({}) String installationId;
+    private @NonNullByDefault({}) String gatewaySerial;
+    private @NonNullByDefault({}) int apiCallLimit;
+    private @NonNullByDefault({}) int bufferApiCommands;
+    private @NonNullByDefault({}) int pollingInterval;
```

Comment:
Why don't you store the configuration object?

Reply:
I don't know exactly what you mean, can you explain to me what I can do?

Reply:
Instead of

```
this.apiKey = config.apiKey;
...
```

You could simply use

```
this.config = config;
```

and later use `this.config.apiKey` instead of `apiKey`. That makes it more clear where the values come from and also reduces code.